src/peft/tuners/lora/config.py

```diff
@@ -130,3 +151,7 @@ def __post_init__(self):
         # if target_modules is a regex expression, then layers_pattern should be None
         if isinstance(self.target_modules, str) and self.layers_pattern is not None:
             raise ValueError("`layers_pattern` cannot be used when `target_modules` is a str.")
+
+        if is_megatron_available():
+            if self.tensor_parallel is not None or self.megatron_config is not None:
+                assert self.tensor_parallel is not None and self.megatron_config is not None
```

Comment:
I don't understand this check, it checks the same thing as the `if` condition? Also, let's not use `assert` but raise proper errors instead.

Reply:
> I don't understand this check, it checks the same thing as the `if` condition? Also, let's not use `assert` but raise proper errors instead.

This check means is that either both of them are None, or neither of item is None.

Reply:
```python
        if is_megatron_available():
            if (self.tensor_parallel is not None or self.megatron_config is not None) and (
                self.tensor_parallel is None or self.megatron_config is None
            ):
                raise ValueError("`tensor_parallel` and `megatron_config` must both have value or neither have value.")
```
Now, it has been modified like this.

Reply:
> 我不明白这个检查，它检查与`if`条件相同的东西？另外，我们不要使用`assert`，而是提出适当的错误。



> I don't understand this check, it checks the same thing as the `if` condition? Also, let's not use `assert` but raise proper errors instead.

Megatron.core is now used by default, based on megatron_config, so this check is no longer needed.